appcelerator/titanium_mobile
# [TIMOB-25498] Deprecate getEventsInYear and getEventsInMonth

**JIRA:** https://jira.appcelerator.org/browse/TIMOB-25498

**Description:**
Deprecate getEventsInYear and  getEventsInMonth from Titanium.Calendar.Calendar

**Test:**

Run the below on both Android and iOS...

1. Add the below XML settings to your project's "tiapp.xml" file.
2. Add the below JavaScript code to your project.
3. Build and run the project.
4. If a calendar permission request dialog is displayed, then authorize it.
5. Look in the log and verify that the following warnings are displayed
```
getEventsInYear(year) has been deprecated in 7.0.0 in favor of getEventsBetweenDates(date1, date2) to avoid platform-differences of the month-index between iOS and Android
getEventsInMonth(year, month) has been deprecated in 7.0.0 in favor of getEventsBetweenDates(date1, date2) to avoid platform-differences of the month-index between iOS and Android
```

---

tiapp.xml

```xml
<?xml version="1.0" encoding="UTF-8"?>
<ti:app xmlns:ti="http://ti.appcelerator.org">
	<android xmlns:android="http://schemas.android.com/apk/res/android">
		<manifest>
			<uses-permission android:name="android.permission.READ_CALENDAR"/>
		</manifest>
	</android>
</ti:app>
```

---

app.js

```javascript
function performTest() {
	Ti.API.info("Calendar test started.");
	var calendarArray = Ti.Calendar.allCalendars;
	if (calendarArray && (calendarArray.length > 0)) {
		var calendar = calendarArray[0];
		calendar.getEventsInYear(2016);
		calendar.getEventsInMonth(2016, 1);
	} else {
		Ti.API.info("Calendar not found.");
	}
}

if (Ti.Calendar.hasCalendarPermissions()) {
	performTest();
} else {
	Ti.Calendar.requestCalendarPermissions(function(e) {
		if (e.success) {
			performTest();
		}
	});
}
```
